internal/gatewayapi/contexts.go

```diff
@@ -417,3 +417,27 @@ func (r *RouteParentContext) HasCondition(route RouteContext, condType gwapiv1.R
 	}
 	return false
 }
+
+// BackendRefContext represents a generic BackendRef object that can reference Gateway objects.
```

Comment:
id suggest this be an interface with 2 methods
* GetBackendRef()
* GetFilters()

and the implementation could look similar to https://github.com/envoyproxy/gateway/blob/c5cd964da2f2ec1951ba72d45b54eef8246e96ee/internal/gatewayapi/contexts.go#L216
```


e.g. reflect.ValueOf(route).Elem().FieldByName("BackendRef")

Reply:
I think the struct would be still needed or am I wrong?